### PR TITLE
FileInbound DSL: Add `recursive` for convenience

### PR DESCRIPTION
**Pull request by artembilan**
Related to https://stackoverflow.com/questions/66171881/how-to-read-nested-txt-file-from-spring-integration-file

The `FileInboundChannelAdapterSpec` can be configured with an external `DirectoryScanner`,
but it sometimes becomes burden for end-users to extract a scanner bean and configure it
with all the required file filters

* Expose a `recursive(boolean)` option for better end-user experience
* Rework `FileTests` for JUnit 5
* Mention a new option in the docs

### CODE REVIEW

File: spring-integration-file/src/main/java/org/springframework/integration/file/dsl/FileInboundChannelAdapterSpec.java [hunk 1]
```diff
@@ -75,6 +76,22 @@ FileInboundChannelAdapterSpec directory(File directory) {
...
+	 * A convenient flag to determine if target message source should use a
+	 * {@link RecursiveDirectoryScanner} or stay with a default one.
+	 * @param recursive to set or not a {@link RecursiveDirectoryScanner}.
+	 * @return the spec.
+	 * @see org.springframework.integration.file.RecursiveDirectoryScanner
+	 * @since 5.5
+	 */
+	public FileInboundChannelAdapterSpec recursive(boolean recursive) {
+		if (recursive) {
+			new DirectFieldAccessor(this.target).setPropertyValue("scanner", new RecursiveDirectoryScanner());
```
Comment (garyrussell) [inline spring-integration-file/src/main/java/org/springframework/integration/file/dsl/FileInboundChannelAdapterSpec.java:89]: Why use recursion when target has `setScanner`?

File: src/reference/asciidoc/whats-new.adoc [hunk 2]
```diff
@@ -58,6 +58,8 @@ In addition, `forRecursion=true` causes the full path to files to be used as the
 IMPORTANT: This means that existing keys in a persistent metadata store will not be found for files beneath the top level directory.
 For this reason, the property is `false` by default; this may change in a future release.
 
+The `FileInboundChannelAdapterSpec` has now a convenient `recursive(boolean)` option instead of requesting an explicit reference to the `RecursiveDirectoryScanner`.
```
Comment (garyrussell) [inline src/reference/asciidoc/whats-new.adoc:61]:
```suggestion
The `FileInboundChannelAdapterSpec` has now a convenient `recursive(boolean)` option instead of requiring an explicit reference to the `RecursiveDirectoryScanner`.
```

File: src/reference/asciidoc/file.adoc [hunk 3]
```diff
@@ -216,6 +215,8 @@ All other sub-directories inclusions and exclusions are based on the target `Fil
 For example, the `SimplePatternFileListFilter` filters out directories by default.
 See https://docs.spring.io/spring-integration/api/org/springframework/integration/file/filters/AbstractDirectoryAwareFileListFilter.html[`AbstractDirectoryAwareFileListFilter`] and its implementations for more information.
 
+NOTE: Starting with version 5.5, the `FileInboundChannelAdapterSpec` of Java DSL has a convenient `recursive(boolean)` option to use the mentioned `RecursiveDirectoryScanner` in the target `FileReadingMessageSource` instead of default one.
```
Comment (garyrussell) [inline src/reference/asciidoc/file.adoc:218]:
```suggestion
NOTE: Starting with version 5.5, the `FileInboundChannelAdapterSpec` of the Java DSL has a convenient `recursive(boolean)` option to use a `RecursiveDirectoryScanner` in the target `FileReadingMessageSource` instead of the default one.
```

File: spring-integration-file/src/main/java/org/springframework/integration/file/dsl/FileInboundChannelAdapterSpec.java [hunk 4]
```diff
@@ -83,14 +100,14 @@ FileInboundChannelAdapterSpec directory(File directory) {
 	 */
 	public FileInboundChannelAdapterSpec scanner(DirectoryScanner scanner) {
 		this.scanner = scanner;
-		this.target.setScanner(scanner);
+
```
Comment (garyrussell) [inline spring-integration-file/src/main/java/org/springframework/integration/file/dsl/FileInboundChannelAdapterSpec.java:103]: Why is this removed?

**Review by garyrussell — CHANGES_REQUESTED** [review]

File: spring-integration-file/src/main/java/org/springframework/integration/file/dsl/FileInboundChannelAdapterSpec.java (hunk 1 above)
Comment (artembilan) [reply spring-integration-file/src/main/java/org/springframework/integration/file/dsl/FileInboundChannelAdapterSpec.java:89]: I believe you mean "reflection".
It is because the `setFilter` and `setLocker` check for the external scanner. See `FileReadingMessageSource.scannerExplicitlySet` property and its usage.
According end-user experience it is a bit cumbersome to move all the fluent filter configuration to the external scanner, when they just want to change the default scanner to recursive one.

I think we can find some other way to do that with some new setter, but I don't think it hurts at the moment to do this reflection once on configuration phase. If you insist, I'll revise it anyway.

Comment (garyrussell) [reply spring-integration-file/src/main/java/org/springframework/integration/file/dsl/FileInboundChannelAdapterSpec.java:89]: No that's fine; yes, I meant reflection 😄

File: spring-integration-file/src/main/java/org/springframework/integration/file/dsl/FileInboundChannelAdapterSpec.java (hunk 4 above)
Comment (garyrussell) [reply spring-integration-file/src/main/java/org/springframework/integration/file/dsl/FileInboundChannelAdapterSpec.java:103]: How is this resolved? How does an external scanner get applied now?

File: spring-integration-file/src/main/java/org/springframework/integration/file/dsl/FileInboundChannelAdapterSpec.java (hunk 1 above)
Comment (artembilan) [reply spring-integration-file/src/main/java/org/springframework/integration/file/dsl/FileInboundChannelAdapterSpec.java:89]: OK. Let me push some fix and we will continue discussion if that!

File: spring-integration-file/src/main/java/org/springframework/integration/file/dsl/FileInboundChannelAdapterSpec.java (hunk 4 above)
Comment (artembilan) [reply spring-integration-file/src/main/java/org/springframework/integration/file/dsl/FileInboundChannelAdapterSpec.java:103]: It is restored in my local copy. 
Will push soon enough!
Sorry for confusion 😄

**Review by garyrussell — APPROVED** [review]
